### PR TITLE
feat: global toleration and affinity documentation.

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Adds a new HowTo documentation explaining how users can define tolerations and affinity configuration for the whole Kubewarden stack.

### CODE REVIEW

File: docs/howtos/production-deployments.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    policy-servers,
+    production,
+    poddisruptionbudget,
+    affinity,
+    limits,
+    tolerations,
+  ]
+---
+
+# Configuring Tolerations and Affinity/Anti-Affinity for the Kubewarden Stack
```
Comment (jhkrug) [inline docs/howtos/production-deployments.md:30]:
```suggestion
# Configuring Tolerations and Affinity/Anti-Affinity for the Kubewarden stack
```

File: docs/howtos/production-deployments.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+Kubewarden provides robust mechanisms to ensure the reliability and proper
+scheduling of its components within a Kubernetes cluster. Starting from version
+1.15 of the Kubewarden Helm charts, two new fields, `tolerations` and
+`affinity`, have been introduced under the `global` value. These fields allow
+users to define Kubernetes tolerations and affinity/anti-affinity settings for
+the Kubewarden stack, including the controller deployment, audit scanner
+cronjob, and the PolicyServer custom resource.
```
Comment (jhkrug) [inline docs/howtos/production-deployments.md:38]:
```suggestion
Kubewarden provides features for reliability and correct
scheduling of its components in a Kubernetes cluster. 

Starting from version
1.15 of the Kubewarden Helm charts, two new fields, `tolerations` and
`affinity`, are introduced under the `global` value. These fields allow
users to define Kubernetes tolerations and affinity/anti-affinity settings for
the Kubewarden stack, including the controller deployment, audit scanner
cronjob, and the PolicyServer custom resource.
```

File: docs/howtos/production-deployments.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      value: "value1"
+      effect: "NoSchedule"
+    - key: "key2"
+      operator: "Equal"
+      value: "value2"
+      effect: "NoExecute"
+```
+
+In this example, the tolerations defined will be applied to the controller
+deployment, audit scanner cronjob, and the PolicyServer custom resource.
```
Comment (jhkrug) [inline docs/howtos/production-deployments.md:62]:
```suggestion
In this example, the tolerations defined are applied to the controller
deployment, audit scanner cronjob, and the PolicyServer custom resource.
```

File: docs/howtos/production-deployments.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+## Affinity/Anti-Affinity
+
+The `affinity` field allows users to define Kubernetes affinity and
+anti-affinity rules for the Kubewarden components. Affinity rules constrain
+pods to specific nodes, while anti-affinity rules prevent pods from being
+scheduled on certain nodes or in close proximity to other pods. These settings
+are crucial for ensuring high availability, fault tolerance, and optimized
```
Comment (jhkrug) [inline docs/howtos/production-deployments.md:70]:
```suggestion
are useful for ensuring high availability, fault tolerance, and optimized
```

File: docs/howtos/production-deployments.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+resource usage within the cluster.
```
Comment (jhkrug) [inline docs/howtos/production-deployments.md:71]:
```suggestion
resource usage in a cluster.
```

File: docs/howtos/production-deployments.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            - key-2
+```
+
+In this example, the affinity rules will be applied to the controller
+deployment, audit scanner cronjob, and the PolicyServer custom resource.
+
+The previous affinity configuration available in the `kubewarden-default` Helm
+chart, which was used to define the affinity configuration for the PolicyServer
+only, has been removed in favor of the global `affinity` field. This change
+simplifies the configuration process by providing a single, unified approach to
```
Comment (jhkrug) [inline docs/howtos/production-deployments.md:127]:
```suggestion
simplifies the configuration process by providing a single approach to
```

File: docs/howtos/production-deployments.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+defining affinity and anti-affinity rules for all Kubewarden components.
+
+:::warning The old `affinity` configuration in the `kubewarden-default` Helm
+chart has been deprecated and removed. Users should now use the
+`global.affinity` field to configure affinity and anti-affinity settings for
+the entire Kubewarden stack. 
+:::
+
+By leveraging the `tolerations` and `affinity` fields, operators can fine-tune
```
Comment (jhkrug) [inline docs/howtos/production-deployments.md:136]:
```suggestion
By using the `tolerations` and `affinity` fields, operators can fine-tune
```

**Review by jhkrug — CHANGES_REQUESTED** [review]
Hi, some minor suggestions if you like them.

File: docs/howtos/production-deployments.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+deployment, audit scanner cronjob, and the PolicyServer custom resource.
+
+The previous affinity configuration available in the `kubewarden-default` Helm
+chart, which was used to define the affinity configuration for the PolicyServer
+only, has been removed in favor of the global `affinity` field. This change
+simplifies the configuration process by providing a single approach to
+defining affinity and anti-affinity rules for all Kubewarden components.
+
+:::warning The old `affinity` configuration in the `kubewarden-default` Helm
+chart has been deprecated and removed. Users should now use the
```
Comment (flavio) [inline docs/howtos/production-deployments.md:133]: Something cannot be deprecated and removed at the same time. AFAIK we went straight to remove it

```suggestion
:::warning The old `affinity` configuration in the `kubewarden-default` Helm
chart has been removed. Users should now use the
```

**Review by flavio — COMMENTED** [review]
Looks good to me, thanks for the update. I left one minor suggestion

**Review by viccuad — APPROVED** [review]
LGTM modulo Flavio's suggestions.

**Pull request merged**